Render the site menu based on user configuration

lpancescu opened the pull request:
Hugo supports menus defined in config.toml, as well as the frontmatter.
We can iterate through Sites.Menu.<menu_name> to generate the site menu,
although my implementation doesn't support nested menus.

I also changed the example site to use the new menu implementation,
while keeping the previous menu ordering and generated CSS classes.

Please see https://gohugo.io/extras/menus/ for details.

nishanths commented on layouts/partials/header.html:
```diff
@@ -19,9 +19,10 @@ <h3 class="self-intro">{{ .Site.Params.selfintro }}</h3>
-                    <li class="nav-blog"><a href="/blog/">Blog</a></li>
-                    <li class="nav-about"><a href="/about/">About</a></li>
-                    <li class="nav-code"><a href="/code/">Code</a></li>
+                    {{ $currentPage := . }}
```
Is this assignment useful? Can we remove it if it's not?

lpancescu commented:
@nishanths I removed the unused assignment and refactored the example site to define the menu structure just in `config.toml`, instead of having that spreaded in different content files. Thanks!
